learningeconomy/LearnCard
# [LC-158] Certificate Display Card

[![workerB](https://img.shields.io/endpoint?url=https%3A%2F%2Fworkerb.linearb.io%2Fv2%2Fbadge%2Fprivate%2FU2FsdGVkX19ZqAkpy5N6Re87rhZX4KA06D5V1kILso%2Fcollaboration.svg%3FcacheSeconds%3D60)](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=uZX1IbX)
# Overview

#### 🎟 Relevant Jira Issues

[LC-158](https://welibrary.atlassian.net/browse/LC-158)

#### 📚 What is the context and goal of this PR?
We're introducing a certificate display type for official sorts of uses. What does that mean? NEW UI

- Boosts with certificate display types should now be displayed with a brand new UI
- The outline colors are the category's 500 value
- The background for the back face and the "issuer stamp" (spiky circle thing) should be the category's 700 value
- When viewing the boost there's a ... button that allows you to share your boost
- The boost distinguishes between Self Verified (issued to self), Trusted Verifier (in the [trusted apps registry](https://github.com/learningeconomy/registries/blob/main/learncard/trusted-app-registry.json) with `app.isTrusted` set to true), Unknown Verifier (not in the registry), and Untrusted Verifier (in the registry with `app.isTrusted` set to false
- In the manage boosts tab, the boost will display up to 3 recipients' profile images (with "+X" if applicable)

(Note: Skills related items are not part of this PR)

#### 🥴 TL; RL:
New UI for new boost type!

#### 💡 Feature Breakdown (screenshots & videos encouraged!)
Preview (front side w/ background)
![image](https://github.com/learningeconomy/LearnCard/assets/63376352/cbfd9905-c08f-4cba-9805-bc3b230fbf14)

Back Side
![image](https://github.com/learningeconomy/LearnCard/assets/63376352/ba2e7350-8821-4421-aab1-3a51cd0020ee)

Manage Tab -> multiple recipients
![image](https://github.com/learningeconomy/LearnCard/assets/63376352/14818847-9912-43e8-bb42-2ab340b88bb1)

Issued, self verified (with ... menu)
![image](https://github.com/learningeconomy/LearnCard/assets/63376352/ba98cdda-5d65-488f-8f7f-ca9d8301b15d)

Untrusted Verifier
![image](https://github.com/learningeconomy/LearnCard/assets/63376352/91d36fb9-ddd6-47c0-93bc-a6182da1aa80)

Trusted Verifier 
![image](https://github.com/learningeconomy/LearnCard/assets/63376352/f50417fc-e000-4850-9f9c-11fd240c6cb3)

#### 🛠 Important tradeoffs made:

#### 🔍 Types of Changes


#### 💳 Does This Create Any New Technical Debt? ( If yes, please describe and [add JIRA TODOs](https://welibrary.atlassian.net/jira/software/projects/WE/boards/2) )

# Testing

#### 🔬 How Can Someone QA This?


**Use `learncardapp` branch `lc-157`** - these two are very tied together

- Go through the Issue Boost flow and select the Certificate display type (only available for some categories)
- Preview the boost and verify that it all looks good and matches the design / user stories
  - double check that badge display still works fine
- Boosts that are actually issued are a bit trickier since we don't actually store that display type yet. The easiest way to do this is to search for `const isCertificate` in learncardapp and react-learn-card. Then just set all of those to true and everything will appear in certificate display mode
  - components in learncardapp + learn-card-base: BoostManagedCard, BoostPreview, NonBoostPreview, VCDisplayCardWrapper2, VCModal,
  - There's also a displayType condition circa line 153 in VCDisplayCard2 in react-learn-card
- open a bunch of credentials in your wallets. Try all the categories and configurations. Make sure they all look good and work as expected
  - issued credentials should have the ... button
- Go to the Managed tab for a boost you've issued to multiple people. Open that one up, you should see multiple users' profile pics + "Issued to X people"
- Test self verified / trusted verifier / untrusted verifier / unknown verifier
  - You can update the "Kyle Dev" entry however you like
  - `did` and `app.isTrusted` are the relevant fields

#### 📱 🖥 Which devices would you like help testing on?


#### 🧪 Code Coverage


# Documentation

#### 📜 Gitbook


#### 📊 Storybook



# ✅ PR Checklist

### 🚀 Ready to squash-and-merge?:
